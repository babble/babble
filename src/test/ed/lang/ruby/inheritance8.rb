
def silly( n )
  puts "_"
  puts self.A
  puts n
end

class Lower

  A = 5
  
  attr_accessor :all
  
  def initialize
    @all = [ "1" ]
  end
  
  def a
    return @all
  end
  
  def self.A
    return 111
  end
  
  def self.funny
    puts "funny"
  end
  
end

class Base < Lower
  
  silly( "2" )

  def blah(z)
    @z = z
  end

end

class A < Base
  def go
    blah( 5 )
  end

  def p
    puts @all.length
    puts @z
  end
end

class B < Base
  def go
    blah( 6 )
  end

  def p
    puts @all.length
    puts @z
  end

  def self.blah
    puts "B.self.blah"
    funny
  end
    
end

a = A.new
a.go
a.p

b = B.new
b.go
b.p

a.p


B.blah
