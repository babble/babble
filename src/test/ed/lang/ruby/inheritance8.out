_
111
2
5
6
5
